persistence/src/test/java/cz/fi/muni/pa165/dao/SportDaoImplTest.java

```diff
@@ -1,8 +1,113 @@
 package cz.fi.muni.pa165.dao;
 
+import cz.fi.muni.pa165.entity.Sport;
+import org.junit.Rule;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
 /**
  * @author Petra Halová
  */
 public class SportDaoImplTest extends BaseDaoImplTest {
 
+    @Autowired
+    SportDao sportDao;
+
+    private Sport sport1;
+    private Sport sport2;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeMethod
+    public void setUp() {
+        sport1 = new Sport();
+        sport1.setName("Pole dance");
+
+        sport2 = new Sport();
+        sport2.setName("chess");
+
+        sportDao.create(sport1);
+        sportDao.create(sport2);
+    }
+
+    /*
+    @Test
+    public void testCreateSportWithNullName(){
+        Sport sportWithNullName = new Sport();
+        sportWithNullName.setName(null);
+        expectedException.expect(IllegalArgumentException.class);
+        sportDao.create(sportWithNullName);
+    }
+*/
+    @Test
+    public void testDeleteSport() {
+
+        assertThat(sportDao.findById(sport1.getId())).isNotNull();
+        assertThat(sportDao.findById(sport2.getId())).isNotNull();
+
+        sportDao.delete(sport1);
+
+        assertThat(sportDao.findById(sport2.getId())).isNotNull();
+        assertThat(sportDao.findById(sport1.getId())).isNull();
+    }
+
+    /*
+    @Test
+    public void testDeleteNullSport() {
+        expectedException.expect(IllegalArgumentException.class);
+        sportDao.delete(null);
+    }
+*/
+    @Test
+    public void testDeleteSportWithNullId() {
+        sport1.setId(null);
+        expectedException.expect(IllegalArgumentException.class);
```

Comment:
tento kod nevyhodi vyjimku.
`Throws: IllegalArgumentException - if the instance is not an entity or is a detached entity`